api/handlers/service_instance_test.go

```diff
@@ -73,6 +73,294 @@ var _ = Describe("ServiceInstance", func() {
+				"foo": "bar",
+			}, nil)
+
+			reqPath += "/service-instance-guid/credentials"
+		})
+
+		It("gets the service instance credentials", func() {
+			Expect(serviceInstanceRepo.GetServiceInstanceCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualServiceInstanceGUID := serviceInstanceRepo.GetServiceInstanceArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualServiceInstanceGUID).To(Equal("service-instance-guid"))
+
+			Expect(serviceInstanceRepo.GetServiceInstanceCredentialsCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualInstanceGUID := serviceInstanceRepo.GetServiceInstanceCredentialsArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualInstanceGUID).To(Equal("service-instance-guid"))
+
+			Expect(rr).Should(HaveHTTPStatus(http.StatusOK))
+			Expect(rr).To(HaveHTTPHeaderWithValue("Content-Type", "application/json"))
+			Expect(rr).To(HaveHTTPBody(SatisfyAll(
+				MatchJSONPath("$.foo", "bar"),
+			)))
+		})
+
+		When("the service instance is not user-provided", func() {
+			BeforeEach(func() {
+				serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+					GUID: "service-instance-guid",
+					Type: korifiv1alpha1.ManagedType,
+				}, nil)
+			})
+
+			It("returns a 404 Not Found error", func() {
+				expectNotFoundError("Service Instance")
+			})
+		})
+
+		When("the service instance does not have credentials", func() {
+			BeforeEach(func() {
+				serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{
+					GUID: "service-instance-guid",
+					Type: korifiv1alpha1.UserProvidedType,
+				}, nil)
+
+				serviceInstanceRepo.GetServiceInstanceCredentialsReturns(map[string]any{}, apierrors.NewNotFoundError(nil, repositories.ServiceInstanceResourceType))
+			})
+
+			It("returns 404 Not Found", func() {
+				_, actualAuthInfo, actualInstanceGUID := serviceInstanceRepo.GetServiceInstanceCredentialsArgsForCall(0)
+				Expect(actualAuthInfo).To(Equal(authInfo))
+				Expect(actualInstanceGUID).To(Equal("service-instance-guid"))
+				expectNotFoundError("Service Instance")
+			})
+		})
+
+		When("getting the service instance fails with not found", func() {
+			BeforeEach(func() {
+				serviceInstanceRepo.GetServiceInstanceReturns(
+					repositories.ServiceInstanceRecord{},
+					apierrors.NewNotFoundError(nil, repositories.ServiceInstanceResourceType),
+				)
+			})
+
+			It("returns 404 Not Found", func() {
+				expectNotFoundError("Service Instance")
+			})
+		})
+
+		When("getting the service instance fails with forbidden", func() {
+			BeforeEach(func() {
+				serviceInstanceRepo.GetServiceInstanceReturns(
+					repositories.ServiceInstanceRecord{},
+					apierrors.NewForbiddenError(nil, repositories.ServiceInstanceResourceType),
+				)
+			})
+
+			It("returns 404 Not Found", func() {
+				expectNotFoundError("Service Instance")
+			})
+		})
+
+		When("the unmarshaling of the secret fails", func() {
```

Comment:
See the comment on the "not found"  error above. The handler does not handle unprocessable entity error in any particular way, it simply returns it. Another context with this particular error does not bring any more value. Therefore I would propose deleting at in order to keep the test simple